pyproject.toml

```diff
@@ -4,12 +4,27 @@ requires = [
     "setuptools",
     "wheel",
     "Cython>=0.28.5",
-    "numpy==1.13.3; python_version=='3.6' and platform_system!='AIX' and platform_python_implementation == 'CPython'",
-    "numpy==1.14.0; python_version=='3.6' and platform_system!='AIX' and platform_python_implementation != 'CPython'",
-    "numpy==1.14.5; python_version=='3.7' and platform_system!='AIX'",
-    "numpy==1.17.3; python_version>='3.8' and platform_system!='AIX'",
-    "numpy==1.16.0; python_version=='3.6' and platform_system=='AIX'",
-    "numpy==1.16.0; python_version=='3.7' and platform_system=='AIX'",
-    "numpy==1.17.3; python_version>='3.8' and platform_system=='AIX'",
+
+    # PyPy needs numpy >= 1.14.0
+    # platform_python_implementation!='CPython' not needed >= Python 3.7 which is numpy 1.14.5
+    "numpy==1.13.3; python_version=='3.6' and platform_machine!='aarch64' and platform_system!='AIX' and platform_python_implementation=='CPython'",
+    "numpy==1.14.0; python_version=='3.6' and platform_machine!='aarch64' and platform_system!='AIX' and platform_python_implementation!='CPython'",
+
+    # AIX needs numpy >= 1.16.0
+    # platform_system!='AIX' not needed >= Python 3.8 which is numpy 1.17.3
+    "numpy==1.16.0; python_version=='3.6' and platform_machine!='aarch64' and platform_system=='AIX'",
+    "numpy==1.16.0; python_version=='3.7' and platform_machine!='aarch64' and platform_system=='AIX'",
+
+    # ARM needs numpy >= 1.19.0
+    # platform_machine!='aarch64' not needed >= Python 3.9 which is numpy 1.19.3
+    "numpy==1.19.0; python_version=='3.6' and platform_machine=='aarch64'",
+    "numpy==1.19.0; python_version=='3.7' and platform_machine=='aarch64'",
+    "numpy==1.19.0; python_version=='3.8' and platform_machine=='aarch64'",
+
+    # default numpy requirements
+    "numpy==1.14.5; python_version=='3.7' and platform_machine!='aarch64' and platform_system!='AIX'",
+    "numpy==1.17.3; python_version=='3.8' and platform_machine!='aarch64'",
+    "numpy==1.19.3; python_version=='3.9'",
+
```

Comment:
What about using [`oldest-supported-numpy`](https://pypi.org/project/oldest-supported-numpy/)?

It would simplify the `pyproject.toml` file quite a lot.

WDYT @thomasjpfan and @ogrisel?

Reply:
That seems like a pretty interesting idea to me, but we'd need to upstream all the changes that were made here (existing pypy constraints and added ARM constraints)

Reply:
Let us accept this solution for now then and keep us track of `oldest-supported-numpy` to see if we could use it in the future.

Reply:
oldest-supported-numpy (implicitly by numpy developers) is not necessarily the same as oldest-supported-numpy by scikit-learn developers.

I think we are rather conservative, so the two might match, but we would have to discuss that with a dedicated issue.

Let's not block this PR further but feel free to open an issue on this.

Reply:
Do you know where the pypy version constraints come from? Looking at PyPI I can see the ARM wheels so I can create a PR for that, but I don't know if I should upstream the PyPy requirement of `1.14.0`

Reply:
Not sure either.

Reply:
See https://github.com/scikit-learn/scikit-learn/pull/11010#issuecomment-396397514. We had 7 failures with `numpy==1.13.3` and none with `numpy==1.14.0` for `PyPy`.

I do not think that we could upstream such dependencies since as @ogrisel pointed out:  "`oldest-supported-numpy` is not necessarily the same as `oldest-supported-numpy` by `scikit-learn` developers."

Sorry for the noise!